app/support/reservations/validations.rb

```diff
@@ -34,6 +35,12 @@ def starts_before_ends
     end
   end
 
+  def starts_before_cutoff
+    if product.cutoff_time && reserve_start_at && in_the_future? && reserve_start_at_changed?
+      errors.add("reserve_start_at", "must be before the cutoff time for this instrument") if reserve_start_at < Time.now + product.cutoff_time.hours
```

Comment:
Or `product.cutoff_time.hours.from_now`. 

I like the idea of localizing the error. That also should allow a test for `expect(reservation.errors).to be_added(:reserve_start_at, :after_cutoff)` to make sure that it's invalid for the right reason.

I also don't love the conditional setup, especially since the important one probably falls off to the right of the screen (might be better after localization) and we also have a conditional in the `before_action` declaration. 

I'm not sure the best setup is. I think `reserve_start_at` and `reserve_start_at_changed?` could move to the `before_action`. Maybe `in_the_future?` as well. Those are sort of the should the validation run, and then `product.cutoff_time` and the time check are the validation itself. That might be the wrong line, though, so I'm open to ideas.